src/overlays/actors/ovl_En_Butte/z_en_butte.c

```diff
@@ -21,41 +45,376 @@ const ActorInit En_Butte_InitVars = {
     (ActorFunc)EnButte_Update,
     (ActorFunc)EnButte_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Butte/func_809CD070.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Butte/func_809CD0F8.s")
+typedef struct EnButteFlightParams {
+    /* 0x00 */ s16 minTime;
+    /* 0x02 */ s16 maxTime;
+    /* 0x04 */ f32 speedXZTarget;
+    /* 0x08 */ f32 speedXZScale;
+    /* 0x0C */ f32 speedXZStep;
+    /* 0x10 */ s16 rotYStep;
+} EnButteFlightParams; // size = 0x14
+
+static EnButteFlightParams sFlyAroundParams[] = {
+    { 5, 35, 0.0f, 0.1f, 0.5f, 0 },
+    { 10, 45, 1.1f, 0.1f, 0.25f, 1000 },
+    { 10, 40, 1.5f, 0.1f, 0.3f, 2000 },
+};
+static EnButteFlightParams sFollowLinkParams[] = {
+    { 3, 3, 0.8f, 0.1f, 0.2f, 0 },
+    { 10, 20, 2.0f, 0.3f, 1.0f, 0 },
+    { 10, 20, 2.4f, 0.3f, 1.0f, 0 },
+};
+
+extern AnimationHeader D_05002470;
+extern SkeletonHeader D_050036F0;
+
+void EnButte_SelectFlightParams(EnButte* this, EnButteFlightParams* flightParams) {
+    if (this->flightParamsIdx == 0) {
+        if (Math_Rand_ZeroOne() < 0.6f) {
+            this->flightParamsIdx = 1;
+        } else {
+            this->flightParamsIdx = 2;
+        }
+    } else {
+        this->flightParamsIdx = 0;
+    }
+
+    this->timer = Math_Rand_S16Offset(flightParams->minTime, flightParams->maxTime);
+}
+
+static f32 sTransformationEffectScale = 0.0f;
+static s16 sTransformationEffectAlpha = 0;
+
+void EnButte_ResetTransformationEffect(void) {
+    sTransformationEffectScale = 0.0f;
+    sTransformationEffectAlpha = 0;
+}
+
+void EnButte_UpdateTransformationEffect(void) {
+    sTransformationEffectScale += 0.003f;
+    sTransformationEffectAlpha += 4000;
+}
+
+void EnButte_DrawTransformationEffect(EnButte* this, GlobalContext* globalCtx) {
+    static Vec3f D_809CE3C4 = { 0.0f, 0.0f, -3.0f };
+
+    Vec3f sp5C;
```

Comment:
We don't normally space out static and local variables
```suggestion
    Vec3f sp5C;
```